api/query.go

```diff
@@ -57,32 +57,25 @@ func (e *Endpoint) slice(ctx *gin.Context, query SliceQuery) {
 	writeResponse(ctx, metadata, data)
 }
 
-func sliceParseGetReq(ctx *gin.Context) (*SliceQuery, error) {
-	var query SliceQuery
-
-	err := json.Unmarshal([]byte(ctx.Query("query")), &query)
-	if err != nil {
-		return nil, err
-	}
-
-	if err = binding.Validator.ValidateStruct(&query); err != nil {
-		return nil, err
+func parseGetRequest(ctx *gin.Context, v interface{}) error {
+	if err := json.Unmarshal([]byte(ctx.Query("query")), v); err != nil {
```

Comment:
I think I found it more readable when err assignment wasn't inside if.
Now the actual action - unmarshalling - seems more like a side-effect and err - the actual result value.

Same to other 2 uses.